src/goesproc/handler_goesr.cc

```diff
@@ -98,6 +98,49 @@ void GOESRImageHandler::handle(std::shared_ptr<const lrit::File> f) {
   fb.region = details.region;
   fb.channel = details.channel;
 
+  // Process image data function
+  if (f->hasHeader<lrit::ImageDataFunctionHeader>()) {
+    auto h = f->getHeader<lrit::ImageDataFunctionHeader>();
+
+    // Sample IDF (ABI Channel 8), output with lritdump -v
+    //
+    // Image data function (3):
+    //  Data:
+    //    $HALFTONE:=8
+    //    _NAME:=toa_brightness_temperature
+    //    _UNIT:=K
+    //    255:=138.0500
+    //    254:=138.7260
+    //    253:=139.4020
+    //    252:=140.0780
+    //    251:=140.7540
+    // [...]
+    //    5:=307.0494
+    //    4:=307.7254
+    //    3:=308.4014
+    //    2:=309.0774
+    //    1:=309.7534
+    //    0:=310.4294
+
+    const auto str = std::string((const char*) h.data.data(), h.data.size());
+    std::istringstream iss(str);
+    std::string line;
+
+    unsigned int ki;
+    float vf;
+
+    while (std::getline(iss, line, '\n')) {
+      std::istringstream lss(line);
+      std::string k, v;
+      std::getline(lss, k, '=');
+      std::getline(lss, v, '=');
+      k.erase(k.end() - 1);
+      ki = atoi(k.c_str());
+      vf = atof(v.c_str());
+      imageDataFunction_[ki] = vf;
+    }
```

Comment:
I think this should do a bit more detection given the sample data. For the first couple of lines atoi will return 0 or -1 or something invalid/undefined. If they are reordered and appear at the end of the IDF then it may overwrite valid data that preceded it.